cmake/patches/onnx/onnx.patch

```diff
@@ -469,3 +469,561 @@ index 2c63c910..b0e4c32d 100644
++            if (shape_dim.has_dim_value() && current_dim->has_dim_value() &&
++                shape_dim.dim_value() > current_dim->dim_value() && current_dim->dim_value() == 1) {
++              current_dim->set_dim_value(shape_dim.dim_value());
++            }
++          }
++        }
++        index += 2; // skip the rest of dots
++        num_illegal_char += 3;
++        continue;
++
++      } else if (!is_letter(term[index])) {
++        num_illegal_char += 1;
++        continue;
++      }
++
++      const auto inserted = label_maps.insert({term[index], num_labels}).second;
++      if (inserted) {
++        *dims_value.add_dim() = shape.dim(index + ellipsis_dims - num_illegal_char);
++        ++num_labels;
++      } else {
++        repeated_labels.insert(term[index]);
++      }
++    }
++
+     if (ellipsis_index != std::string::npos) {
+       // If there is an ellipsis, the number of dimensions it represents
+       // must be total dim - letter dimensions
+       if (num_ellipsis == 0) {
+-        if (rank + 3 < term.size()) {
++        if (rank < term_size) {
+           fail_shape_inference("Ellipsis represents incompatible dimensions.");
+         }
+-        num_ellipsis_indices = rank - term.size() + 3;
++        num_ellipsis_indices = rank - term_size;
+       } else { // ellipsis has been seen before. Check that if dimensions
+                // are compatible
+-        if (num_ellipsis_indices != rank - term.size() + 3) {
++        if (num_ellipsis_indices != rank - term_size) {
+           fail_shape_inference("Ellipsis represents incompatible dimensions.");
+         }
+       }
+       num_ellipsis++;
+     } else {
+-      if (rank != term.size()) {
++      if (rank != term_size) {
+         fail_shape_inference("Rank of input ", num_operands, " does not match the equation indices.");
+       }
+     }
+     num_operands++;
+   }
+ 
+-  if (numInputs != num_operands) {
++  if (num_inputs != num_operands) {
+     fail_shape_inference("Number of input tensors does not match the operands in the equation.");
+   }
+ 
+-  const size_t number_of_letters = 26;
+-  size_t num_letter_occurrences[number_of_letters] = {0};
+   // Parse the provided right-hand side
+   if (mid_index != std::string::npos) {
+     std::string right_equation = equation.substr(mid_index + 2);
+     auto right_ellipsis_index = right_equation.find("...");
+-    if (right_ellipsis_index != std::string::npos) { // Right-hand side contains ellipsis
+-      for (size_t i = 0; i < num_ellipsis_indices; ++i) {
+-        output_shape->add_dim();
++
++    for (size_t index = 0; index < right_equation.size(); ++index) {
++      // If there's an ellipsis, add its corresponding dimensions
++      if (index == right_ellipsis_index) {
++        for (size_t i = 0; i < num_ellipsis_indices; i++) {
++          *output_shape.add_dim() = ellipsis_dims_value.dim(i);
++        }
++        index += 2; // skip the rest of dots
++        continue;
+       }
+-    }
```

Comment:
This braces removal seems extra from the original PR.

Reply:
![image](https://github.com/user-attachments/assets/b518f041-8c57-4782-a643-3ca583c937f5)